pkg/detectors/intra42/intra42.go

```diff
@@ -0,0 +1,117 @@
+package intra42
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	regexp "github.com/wasilibs/go-re2"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time
+var _ detectors.Detector = (*Scanner)(nil)
+
+const verifyURL = "https://api.intra.42.fr/oauth/token"
+
+var (
+	defaultClient = common.SaneHttpClient()
+
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives
+	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"intra", "intra42", "client", "secret"}) + `\b(s-s4t2(?:ud|af)-[abcdef0123456789]{64})\b`)
+	idPat  = regexp.MustCompile(detectors.PrefixRegex([]string{"intra", "intra42", "client", "id"}) + `\b(u-s4t2(?:ud|af)-[abcdef0123456789]{64})\b`)
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"intra", "intra42", "secret", "s-s4t2ud-", "s-s4t2af-"}
```

Comment:
suggestion: I suggest using only `s-s4t2ud-` and `s-s4t2af-` as keywords since they are unique and already serve as prefixes. Including `secret` and `intra` could lead to unnecessary matches that will require filtering out through a costly regex operation later.

Reply:
I wasn't sure if the keywords should be words outside of the regex or just any word in the files. Seems its the later, so I will keep only the prefixes as suggested. Thanks!